### PR TITLE
Add local support to state models

### PR DESCRIPTION
**Pull request by iMicknl**
@tetienne I could use some help here in the coming weeks haha, not sure why attrs is not playing nice with me. There is a pytest for local support, which will help in the implementation.

Eventually, I am even thinking if we should make multiple classes like:

`BaseSetup` and `CloudSetup` or `Setup` (which extends from BaseSetup)

### CODE REVIEW

**Comment by tetienne** [comment]
@iMicknl If you tell me how to reproduce your error, I can help you with `attrs`.

**Comment by iMicknl** [comment]
> @iMicknl Mick Vleeshouwer FTE If you tell me how to reproduce your error, I can help you with `attrs`.

See pytest (or CI here).
```
  E   TypeError: __init__() missing 5 required keyword-only arguments: 'mode', 'time_reliable', 'update_status', 'sync_in_progress', and 'type'
```

**Comment by iMicknl** [comment]
Tests pass, however still in doubt if we should not handle this differently with a Setup() interface and a CloudSetup() and LocalSetup() class. We could set the type of API (cloud vs local) response in the `OverkizServer` definition.

Or will this be fine and are we not really using the optional fields?

File: tests/test_client.py [hunk 1]
```diff
@@ -29,21 +29,31 @@ async def test_get_devices_basic(self, client):
             devices = await client.get_devices()
             assert len(devices) == 23
 
+    @pytest.mark.parametrize(
+        "fixture_name, event_length",
+        [
+            ("events.json", 16),
+            # ("local_events.json", 3),
```
[the remark this answers is not in the document]
Comment (iMicknl) [reply tests/test_client.py:36]: Fixed

**Review by vlebourl — APPROVED** [review]

**Pull request merged**